test/objective.jl

```diff
@@ -64,6 +64,26 @@ function objectives_test(ModelType::Type{<:JuMP.AbstractModel}, VariableRefType:
             4x + 1, @inferred JuMP.objective_function(m, AffExprType))
     end
 
+    @testset "Linear objective changes" begin
+        m = ModelType()
```

Comment:
Just use `m = Model()` instead.  Then you don't need the `== VariableRef` below.